src/include/catalog/database_catalog.h

```diff
@@ -529,151 +263,85 @@ class DatabaseCatalog {
-   * @param len length of type in bytes. len should be -1 for varlen types
-   * @param by_val true if type should be passed by value. false if passed by reference
-   * @param type_category category of type
-   */
-  void InsertType(common::ManagedPointer<transaction::TransactionContext> txn, type_oid_t type_oid,
-                  const std::string &name, namespace_oid_t namespace_oid, int16_t len, bool by_val,
-                  postgres::Type type_category);
-
-  /**
-   * Helper function to insert a type into PG_Type and the type indexes
-   * @param txn transaction to insert with
-   * @param internal_type internal type to insert
-   * @param name type name
-   * @param namespace_oid namespace to insert type into
-   * @param len length of type in bytes. len should be -1 for varlen types
-   * @param by_val true if type should be passed by value. false if passed by reference
-   * @param type_category category of type
-   */
-  void InsertType(common::ManagedPointer<transaction::TransactionContext> txn, type::TypeId internal_type,
-                  const std::string &name, namespace_oid_t namespace_oid, int16_t len, bool by_val,
-                  postgres::Type type_category);
-
-  /**
-   * Helper function to query the oid and kind from
-   * [pg_class](https://www.postgresql.org/docs/9.3/catalog-pg-class.html)
-   * @param txn transaction to query
-   * @param namespace_oid the namespace oid
-   * @param name name of the table, index, view, etc.
-   * @return a pair of oid and ClassKind
-   */
-  std::pair<uint32_t, postgres::ClassKind> GetClassOidKind(common::ManagedPointer<transaction::TransactionContext> txn,
-                                                           namespace_oid_t ns_oid, const std::string &name);
-
-  /**
-   * Helper function to query an object pointer form pg_class
-   * @param txn transaction to query
-   * @param oid oid to object
-   * @return pair of ptr to and ClassKind of object requested. ptr will be nullptr if no entry was found for the given
-   * oid
-   */
-  std::pair<void *, postgres::ClassKind> GetClassPtrKind(common::ManagedPointer<transaction::TransactionContext> txn,
-                                                         uint32_t oid);
-
-  /**
-   * Helper function to query a schema pointer form pg_class
-   * @param txn transaction to query
-   * @param oid oid to object
-   * @return pair of ptr to schema and ClassKind of object requested. ptr will be nullptr if no entry was found for the
-   * given oid
-   */
-  std::pair<void *, postgres::ClassKind> GetClassSchemaPtrKind(
-      common::ManagedPointer<transaction::TransactionContext> txn, uint32_t oid);
+  /** @brief Create a column entry in pg_attribute. @see PgAttributeImpl::CreateColumn */
+  template <typename Column, typename ClassOid, typename ColOid>
+  bool CreateColumn(common::ManagedPointer<transaction::TransactionContext> txn, ClassOid class_oid, ColOid col_oid,
+                    const Column &col);
```

Comment:
Does this actually need to be exposed outside of `Core`?  The original intent was that `pg_attribute` was never updated in isolation and that things like "creating a column" were only achievable through providing an updated schema object.  When we get to DDL, exposing this strikes me as an amazing footgun for the cached schema pointer to be out-of-sync with the underlying data.

Reply:
Took away their guns (specifically, removed `CreateColumn` and `DeleteColumns`).

Is exposing `GetColumns` okay? `RecoveryManager` uses it. Leaving unresolved for now.

Reply:
`GetColumns` is fine since it's read-only and I think needed in some other logic as well.

Reply:
Marking resolved